File: src/main/java/gargoyle/ct/pref/impl/prop/CTPrefByteProperty.java
package gargoyle.ct.pref.impl.prop;

import gargoyle.ct.pref.CTPreferencesProvider;
import gargoyle.ct.prop.CTNumberProperty;
import org.jetbrains.annotations.NotNull;

public class CTPrefByteProperty extends CTPrefProperty<Byte> implements CTNumberProperty<Byte> {
    public CTPrefByteProperty(@NotNull CTPreferencesProvider provider, String name) {
        this(provider, name, (byte) 0);
    }

    public CTPrefByteProperty(@NotNull CTPreferencesProvider provider, String name, Byte def) {
        super(Byte.class, provider, name, def);
    }
}

File: src/main/kotlin/gargoyle/ct/pref/impl/prop/CTPrefByteProperty.kt
package gargoyle.ct.pref.impl.prop

import gargoyle.ct.pref.CTPreferencesProvider
import gargoyle.ct.prop.CTNumberProperty

class CTPrefByteProperty @JvmOverloads constructor(
    provider: CTPreferencesProvider,
    name: String,
    def: Byte = 0.toByte()
) : CTPrefProperty<Byte>(
    Byte::class, provider, name, def
), CTNumberProperty<Byte>
